### PR TITLE
#163518689 Create the new article

### PR DESCRIPTION
**Pull request by Musigwa-zz**
#### What does this PR do?
- It helps the logged in user to create his/her new article
#### Description of Task to be completed?
- Secure the endpoint with JsonWebToken (created the checkAuth middleware)
- Create the request handler of the article creation endpoint
#### How should this be manually tested?
- After cloning the repository, check out `ft-create-article-endpoints-163518689` branch
- Make sure to read `README.md` file to get the setup instructions
- Using any `API client tool(usually POSTMAN)`,
- Create the user and login to get the `access-token`
- Create an article using `POST /api/v1/articles` endpoint
- Make sure to pass the token under `request headers`. It should be {`Authorization`: `Bearer <access-token>`} KEY, VALUE PAIR
- the request body should be as mentioned in the `README.md` file under `Endpoints:` ==> `Create Article` subtitle
After the successful article creation, you should get back the article information. with the successful message.

#### Any background context you want to provide?
I have moved the following: `signup.test.js` and `login.test.js` files into `tests/integration` directory to have a clean structure.
#### What are the relevant pivotal tracker stories?
- #163518689

### CODE REVIEW

File: middlewares/checkAuth.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * @param  {object} req - The request object
+ * @param  {object} res - The response object
+ * @param  {function} next - The next handler function in the request pipeline
+ */
+export default (req, res, next) => {
+  const { authorization } = req.headers;
+  try {
+    const [bearer, token] = authorization.split(" ");
+    if (bearer !== "Bearer")
+      return res.status(INTERNAL_SERVER_ERROR).json({ message: "Access denied" });
```
Comment (espoirMur) [inline middlewares/checkAuth.js]: @Musigwa, Bro "Access denied" is not meaningful to users!
You may add 'you are not authorized to perform this action'.

Comment (espoirMur) [reply middlewares/checkAuth.js]: you can check [this ](https://softwareengineering.stackexchange.com/a/185715/281564)for more information about errors messages

Comment (espoirMur) [reply middlewares/checkAuth.js]: And I think passport has better option for protecting routes check with @abayo-luc !

File: tests/integration/article.test.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import chaiHttp from "chai-http";
```
Comment (espoirMur) [inline tests/integration/article.test.js:1]: @Musigwa, bro!
It looks like this feature is not heavily tested.
Would you consider adding more test cases?
Eg : 
- what happens when you are creating an article and title is not provided?
- what happens when you are creating a post and body is not provided?
- what happens when you are creating a post and description is not provided?
- what result are you expecting after creating a post?

**Review by espoirMur — CHANGES_REQUESTED** [review]

**Review by espoirMur — APPROVED** [review]

**Review by niomwungeri-fabrice — CHANGES_REQUESTED** [review]
Please return list tags in create article response @Musigwa

**Review by niomwungeri-fabrice — APPROVED** [review]

**Comment by abayo-luc** [comment]
Your migration are causing issue, especially when someone is trying to run ```sequelize db:migrate:undo```

**Pull request merged**